src/components/AssetSearch/components/GroupedAssetRow.tsx

```diff
@@ -0,0 +1,301 @@
+import { isSome } from '@/lib/utils'
+import { selectRelatedAssetIdsInclusiveSorted } from '@/state/slices/related-assets-selectors'
+import {
+  selectAssets,
+  selectFeeAssetByChainId,
+  selectFeeAssetById,
+  selectMarketDataByAssetIdUserCurrency,
+  selectPortfolioCryptoPrecisionBalanceByFilter,
+  selectPortfolioUserCurrencyBalanceByAssetId,
+} from '@/state/slices/selectors'
+import { useAppSelector, useSelectorWithArgs } from '@/state/store'
+
+type GroupedAssetRowProps = {
+  asset: Asset
+  handleClick: (asset: Asset) => void
+  disableUnsupported?: boolean
+  hideZeroBalanceAmounts?: boolean
+  showPrice?: boolean
+  onLongPress?: (asset: Asset) => void
+}
+
+export const GroupedAssetRow: FC<GroupedAssetRowProps> = ({
+  asset,
+  handleClick,
+  disableUnsupported,
+  hideZeroBalanceAmounts,
+  showPrice,
+  onLongPress,
+}) => {
+  const { isOpen, onToggle } = useDisclosure()
+  const assets = useAppSelector(selectAssets)
+  const {
+    state: { isConnected },
+  } = useWallet()
+
+  const relatedAssetIdsFilter = useMemo(
+    () => ({
+      assetId: asset.assetId,
+      // We want all related assetIds, and conditionally mark the disconnected/unsupported ones as
+      // disabled in the UI. This allows users to see our product supports more assets than they
+      // have connected chains for.
+      onlyConnectedChains: false,
+    }),
+    [asset],
+  )
+  const relatedAssetIds = useSelectorWithArgs(
+    selectRelatedAssetIdsInclusiveSorted,
+    relatedAssetIdsFilter,
+  )
+
+  const relatedAssets = useMemo(() => {
+    return relatedAssetIds.map(assetId => assets[assetId]).filter(isSome)
+  }, [assets, relatedAssetIds])
+
+  const titleColor = useColorModeValue('black', 'white')
+
+  const feeAsset = useAppSelector(state => selectFeeAssetById(state, asset.assetId))
+
+  const primaryAsset = useMemo(() => {
+    const assetWithoutChainSuffix = relatedAssets.find(
+      relatedAsset =>
+        relatedAsset?.name.includes(' on ') || feeAsset?.assetId === relatedAsset?.assetId,
+    )
+    return assetWithoutChainSuffix || asset
+  }, [relatedAssets, feeAsset, asset])
+
+  const marketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, primaryAsset.assetId ?? ''),
+  )
+
+  const primaryAssetNameWithoutChain = useMemo(() => {
+    return primaryAsset.name.split(' on ')[0]
+  }, [primaryAsset.name])
+
+  const relatedFiatBalances = useAppSelector(state =>
+    relatedAssetIds.map(
+      id => selectPortfolioUserCurrencyBalanceByAssetId(state, { assetId: id }) ?? '0',
+    ),
+  )
+  const totalBalanceUserCurrency = useMemo(
+    () => relatedFiatBalances.reduce((sum, v) => bnOrZero(sum).plus(v).toString(), '0'),
+    [relatedFiatBalances],
+  )
+
+  const relatedCryptoBalances = useAppSelector(state =>
+    relatedAssetIds.map(id =>
+      selectPortfolioCryptoPrecisionBalanceByFilter(state, { assetId: id }),
+    ),
+  )
+  const totalBalanceCryptoHuman = useMemo(
```

Comment:
q: couldn't we do this only once with crypto and prorate with market data user currency? That would keep things a lil bit simpler (assuming all variants have the same price over all chains)